src/gar.lua

```diff
@@ -506,6 +506,11 @@ function decreaseDelegateStakeAtGateway(delegateAddress, gateway, quantity, ban)
 		"Quantity is required and must be an integer greater than or equal to 0: " .. quantity
 	)
 	assert(gateway, "Gateway is required")
+	assert(quantity <= delegate.delegatedStake, "Quantity cannot be greater than the delegate's stake")
+	assert(
+		quantity <= gateway.totalDelegatedStake,
+		"Quantity cannot be greater than the gateway's total delegated stake"
+	)
```

Comment:
protect against the computation ever updating stake amounts the scenario we use pcall and don't copy correctly.